File: cpp/se/TransferOp.cpp
#include "se/TransferOp.hpp"

// Static member initialisation.
std::vector<const EnumType *> TransferOp::values;

const TransferOp* TransferOp::QUERY()
{
	static TransferOp op("query");
	return &op;
}

const TransferOp* TransferOp::REQUEST()
{
	static TransferOp op("request");
	return &op;
}

const TransferOp* TransferOp::CANCEL()
{
	static TransferOp op("cancel");
	return &op;
}

const TransferOp* TransferOp::APPROVE()
{
	static TransferOp op("approve");
	return &op;
}

const TransferOp* TransferOp::REJECT()
{
	static TransferOp op("reject");
	return &op;
}


TransferOp::TransferOp (const std::string &op)
    : EnumType (values, op)
{ }

const TransferOp* TransferOp::value (const std::string &name)
{
    return (const TransferOp *)EnumType::value (name, values);
}

void TransferOp::init()
{
    QUERY();
    REQUEST();
    CANCEL();
    APPROVE();
    REJECT();
}

File: src/main/java/com/ausregistry/jtoolkit2/se/TransferOp.java
package com.ausregistry.jtoolkit2.se;

/**
 * An enumeration of the transfer operation types defined for transfer commands
 * in RFC5730.
 */
public enum TransferOp {
    QUERY("query"),
    REQUEST("request"),
    CANCEL("cancel"),
    APPROVE("approve"),
    REJECT("reject");

    private String op;

    TransferOp(String operation) {
        op = operation;
    }

    @Override
    public String toString() {
        return op;
    }
}

